core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -598,6 +598,15 @@ object SparkSubmit extends CommandLineUtils with Logging {
       }
     }
 
+    // In case of shells, spark.ui.showConsoleProgress can be true by default or by user.
+    if (isShell(args.primaryResource)) {
+      if (!sparkConf.contains("spark.ui.showConsoleProgress")) {
+        sysProps("spark.ui.showConsoleProgress") = "true"
+      }
+    } else {
+      sysProps("spark.ui.showConsoleProgress") = "false"
```

Comment:
This is not needed right?

Reply:
This'll prevent user's mistakes by overriding user's `spark.ui.showConsoleProgress=true` with `SparkSubmit` CLI. In case of `SparkSubmit`, only shells will accept user's `spark.ui.showConsoleProgress`.

Reply:
Or, yes. It can be removed because `SparkContext` also accepts user's configuration. In that case, you're right. This is not needed.

Reply:
Originally, I prefer blocking it from here. Please comment me back if this should be removed.

Reply:
I don't see a need to prevent users from forcing this configuration if they want to.

Reply:
Yep. I see. Thanks!